File: src/main/java/edu/austral/starship/base/framework/WindowSettings.java
package edu.austral.starship.base.framework;

import processing.core.PApplet;

public class WindowSettings {
    private final PApplet applet;

    public WindowSettings(PApplet applet) {
        this.applet = applet;
    }

    public WindowSettings setSize(int width, int height) {
        applet.size(width, height);
        return this;
    }

    public WindowSettings enableHighPixelDensity() {
        applet.pixelDensity(2);
        return this;
    }

    public WindowSettings disableCursor() {
        applet.noCursor();
        return this;
    }

    public WindowSettings enableCursor() {
        applet.cursor();
        return this;
    }

    public WindowSettings fullScreen() {
        applet.fullScreen();
        return this;
    }

    public WindowSettings setFrameRate(int rate) {
        applet.frameRate(rate);
        return this;
    }
}

File: src/main/scala/edu/austral/starship/scala/base/framework/WindowSettings.scala
package edu.austral.starship.scala.base.framework

import processing.core.PApplet

case class WindowSettings(applet: PApplet) {
  def setSize(width: Int, height: Int): WindowSettings = {
    applet.size(width, height)
    this
  }

  def enableHighPixelDensity(): WindowSettings = {
    applet.pixelDensity(2)
    this
  }

  def disableCursor(): WindowSettings = {
    applet.noCursor()
    this
  }

  def enableCursor(): WindowSettings = {
    applet.cursor()
    this
  }

  def fullScreen(): WindowSettings = {
    applet.fullScreen()
    this
  }

  def setFrameRate(rate: Int): WindowSettings = {
    applet.frameRate(rate)
    this
  }
}
